Bottom: 01057e4eca656a89d59b6e97526960abd6078360
Top:    d19b13317575e8b0cf469fb4aca35421c5e6221d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 23:56:11 +0000

Prepend when adding things to lists


---

diff --git a/src/hid/step/object3d_step.c b/src/hid/step/object3d_step.c
index b7a5a87..a04d8eb 100644
--- a/src/hid/step/object3d_step.c
+++ b/src/hid/step/object3d_step.c
@@ -291,10 +291,11 @@ object3d_to_step_body_fragment (step_file *step,
           edge = contour->first_edge;
           do
             {
-              edge_loop_edges = g_list_append (edge_loop_edges, GINT_TO_POINTER (ORIENTED_EDGE_IDENTIFIER (edge)));
+              edge_loop_edges = g_list_prepend (edge_loop_edges, GINT_TO_POINTER (ORIENTED_EDGE_IDENTIFIER (edge)));
             }
           while (edge = LNEXT (edge), edge != contour->first_edge);
 
+          edge_loop_edges = g_list_reverse (edge_loop_edges);
           edge_loop = step_edge_loop (step, "NONE", edge_loop_edges);
 
           if (outer_contour)
@@ -302,13 +303,16 @@ object3d_to_step_body_fragment (step_file *step,
           else
             contour->face_bound_identifier = step_face_bound (step, "NONE", edge_loop, true);
 
-          face_contour_list = g_list_append (face_contour_list, GINT_TO_POINTER (contour->face_bound_identifier));
+          face_contour_list = g_list_prepend (face_contour_list, GINT_TO_POINTER (contour->face_bound_identifier));
         }
 
+      face_contour_list = g_list_reverse (face_contour_list);
       face->face_identifier = step_advanced_face (step, "NONE", face_contour_list, face->surface_identifier, !face->surface_orientation_reversed);
-      shell_face_list = g_list_append (shell_face_list, GINT_TO_POINTER (face->face_identifier));
+      shell_face_list = g_list_prepend (shell_face_list, GINT_TO_POINTER (face->face_identifier));
     }
 
+  shell_face_list = g_list_reverse (shell_face_list);
+
   /* Closed shell which bounds the brep solid */
   pcb_shell_identifier = step_closed_shell (step, "NONE", shell_face_list);
   brep_identifier = step_manifold_solid_brep (step, body_name /* This is picked up as the solid body name by Solidworks */, pcb_shell_identifier);
